### PR TITLE
[FIX] account: price at 0, fixed tax price include

### PR DESCRIPTION
**Pull request by Levizar**
## goal:
The aim of this commit is to correct the compute_all calculation
when it is dealing with an article with 0.00 price and having a fixed
tax included in it.

In this situation, this means the tax is directly taken as an expense
by the company that offers it to its client.

This also solves the situation in which we have a 100% discount because
discount is applied before calling `compute_all` by changing the
unit_price.

## Before this commit:
Creating an invoice with a line having a price at 0 or a 100% discount
and a fixed tax price_include resulted in a really weird behavior: the
company had to reimburse the customer for the amount of the tax.
(what a lucky customer :D )

## After this commit:
Customer is sad, he only gets the product for free.
The base is calculated as negative, resulting in a debit amount on the base
income line.
The tax is then counterbalance by it.

The form is displaying a negative untaxed amount, a positive tax and the
total is 0.

ticket: 2660808, 2637512
Community-pr: https://github.com/odoo/odoo/pull/85747

### CODE REVIEW

**Comment by Levizar** [comment]
Last test is currently failing: VAT is 0

**Comment by Levizar** [comment]
Still todo: apply this modif to the pos duplication of compute_all

File: addons/account/tests/test_account_move_out_invoice.py [hunk 1]
```diff
@@ -1775,6 +1775,75 @@ def test_out_invoice_onchange_sequence_number_1(self):
...
+        self.assertRecordValues(invoice, [{
+            'amount_untaxed': -0.05,
+            'amount_tax': 0.05,
+            'amount_total': 0.0,
+        }])
+
+    def test_out_invoice_line_taxes_fixed_price_include_free_product(self):
+        ''' Check that fixed tax include are correctly computed while the price_unit is 0
+        '''
+        # please ensure this test keeps to be consistent with
```
Comment (epictete) [inline addons/account/tests/test_account_move_out_invoice.py:1810]:
```suggestion
        # please ensure this test remains consistent with
```
Just a minor cosmetic change (same for `test_sale_order.py`).

**Review by epictete — APPROVED** [review]
LGTM :+1:

File: addons/sale/tests/test_sale_order.py [hunk 2]
```diff
@@ -8,7 +8,6 @@
 
 from .test_sale_common import TestCommonSaleNoChart
 
-
```
Comment (Feyensv) [inline addons/sale/tests/test_sale_order.py:11]: useless diff to remove :eyes:

**Review by Feyensv — APPROVED** [review]

**Review by smetl — COMMENTED** [review]
@robodoo r+

**Comment by Levizar** [comment]
> Still todo: apply this modif to the pos duplication of compute_all

Verif done: PoS was already doing the job correctly

**Comment by Levizar** [comment]
> Still todo: apply this modif to the pos duplication of compute_all

Verif done: PoS was already doing the job correctly

**Pull request closed without merging**